warehouse/query-core/src/main/java/datawave/query/jexl/visitors/JexlFormattedStringBuildingVisitor.java

```diff
@@ -0,0 +1,762 @@
+        } catch (StackOverflowError e) {
+            
+            throw e;
+        }
+        return formatBuiltQuery(s);
+    }
+    
+    /**
+     * Build a String that is the equivalent JEXL query.
+     *
+     * @param script
+     *            An ASTJexlScript
+     * @return
+     */
+    public static String buildQueryWithoutParse(JexlNode script) {
+        return buildQueryWithoutParse(script, false);
+    }
+    
+    public Object visit(ASTOrNode node, Object data) {
+        StringBuilder sb = (StringBuilder) data;
+        int numChildren = node.jjtGetNumChildren();
+        JexlNode parent = node.jjtGetParent();
+        boolean wrapIt = false;
+        
+        if (!(parent instanceof ASTReferenceExpression || parent instanceof ASTJexlScript || parent instanceof ASTOrNode || numChildren == 0)) {
+            wrapIt = true;
+            sb.append("(");
+        }
+        
+        Collection<String> childStrings = (sortDedupeChildren) ? new TreeSet<>() : new ArrayList<>(numChildren);
+        StringBuilder childSB = new StringBuilder();
+        for (int i = 0; i < numChildren; i++) {
+            node.jjtGetChild(i).jjtAccept(this, childSB);
+            childStrings.add(childSB.toString());
+            childSB.setLength(0);
+        }
+        sb.append(String.join(" || " + NEWLINE, childStrings));
+        
+        if (wrapIt)
+            sb.append(")");
+        
+        return data;
+    }
+    
+    public Object visit(ASTAndNode node, Object data) {
+        StringBuilder sb = (StringBuilder) data;
+        boolean needNewLines = needNewLines(node);
+        int numChildren = node.jjtGetNumChildren();
+        JexlNode parent = node.jjtGetParent();
+        boolean wrapIt = false;
+        
+        if (!(parent instanceof ASTReferenceExpression || parent instanceof ASTJexlScript || parent instanceof ASTAndNode || numChildren == 0)) {
+            wrapIt = true;
+            sb.append("(");
+        }
+        
+        Collection<String> childStrings = (sortDedupeChildren) ? new TreeSet<>() : new ArrayList<>(numChildren);
+        Collection<String> childStringsFormatted = (sortDedupeChildren) ? new TreeSet<>() : new ArrayList<>(numChildren);
+        StringBuilder childSB = new StringBuilder();
+        for (int i = 0; i < numChildren; i++) {
+            node.jjtGetChild(i).jjtAccept(this, childSB);
+            childStrings.add(childSB.toString());
+            childSB.setLength(0);
+        }
+        // If needNewLines is false, we should remove the new lines added to the child strings
+        for (String childString : childStrings) {
+            childStringsFormatted.add(needNewLines ? childString : childString.replace(NEWLINE, ""));
+        }
+        sb.append(String.join(" && " + (needNewLines ? NEWLINE : ""), childStringsFormatted));
+        
+        if (wrapIt)
+            sb.append(")");
+        
+        return data;
+    }
+    
+    public Object visit(ASTEQNode node, Object data) {
```

Comment:
I believe almost all of the methods below this point are unchanged from the JexlStringBuildingVisitor.  It would be better to extend that class and then we would not need all of these copied methods.